File: Data Structures and Algorithms/4. Recursion/Kunal/Pattern/P3.cpp
#include <iostream>
using namespace std;

void Triangle(int rows, int cols) {
    if(rows == 0)
        return;
    if(cols < rows){
        Triangle(rows, cols + 1);
        cout << "* ";
    } else {
        Triangle(rows - 1, 0);
        cout << "\n";
    }
}

int main() {
    int rows = 4;
    Triangle(rows, 0);
    return 0;
}

File: Data Structures and Algorithms/4. Recursion/Kunal/Pattern/P3.java
/*

*
* *
* * *
* * * *
* * * * *
* * * * * *
* * * * * * *

*/

import java.util.*;
class P3 {
    public static void Triangle(int rows, int cols) {
        if(rows == 0)
            return;
        if(cols < rows) {
            Triangle(rows, cols + 1);
            System.out.print("* ");
        } else {
            Triangle(rows - 1, 0);
            System.out.println();
        }
    }

    public static void main(String[] args) {
        int N = 7;
        Triangle(N, 0);
    }
}
